### PR TITLE
ReduceLROnPlateau bug fix

### PR DESCRIPTION
**Pull request by SkafteNicki**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?   
- [x] Did you write any new necessary tests?  
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CHANGELOG.md)?

## What does this PR do?
Fixes https://github.com/PyTorchLightning/pytorch-lightning/issues/1114
During the latest revoke of the learning rate scheduler interface (by me) a small bug related to the `ReduceLROnPlateau` type of scheduler was made. This PR solves that bug and adds a test to check for this in the future.

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
@SkafteNicki pls add note to changelog... then we can go :]

**Comment by SkafteNicki** [comment]
@Borda I have update CHANGELOG now, should be ready for merge now
